pages/database-management/monitoring.mdx

```diff
@@ -510,4 +510,70 @@ to the one below.
 Memgraph also supports exporting metrics in the
 [Prometheus](https://prometheus.io/) format. For more information on how to set
 up the Prometheus exporter, refer to the [Prometheus
-exporter](https://github.com/memgraph/prometheus-exporter) repository.
\ No newline at end of file
+exporter](https://github.com/memgraph/prometheus-exporter) repository.
+
+## Session trace
+
+Session trace enables you to profile the execution of your whole session. 
+In the database context, a session refers to a temporary, interactive connection between a user (or application)
+and the database, used to execute transactions and queries. Session trace therefore serves as an additional log file
+where the following information is logged:
+- session UUID
+- transaction ID of the current query
+- name of the user executing the query
+- name of the executing query
+- explain plan of the executing query
+- query planning start timestamp
+- query planning end timestamp
+- query planning execution time
+- query execution start timestamp
+- query execution end timestamp
+- query execution time
+- query commit start timestamp
+- query commit end timestamp
+- query commit execution time
+- profile plan of the executing query
+- query summary metadata
+
+This additional log data serves two purposes:
+- it profiles the whole workload of the session, therefore increasing visibility of the overall bottlenecks during execution
+- it does not congest the Memgraph system log with overloading information
+
+The session trace is turned on with the following query:
+```cypher
+SET SESSION TRACE ON;
+```
+
+The result retrieved from the query is the string value of the session UUID being tracked:
+
+ | session uuid                           |
+ | -------------------------------------- |
+ | "de9c907b-6675-40bd-bf09-d4ce7b24f22d" |
+
+The destination of the session trace log is `{--query-log-directory}/{session uuid}.log`
+Where the `--query-log-directory` is a flag that can be set during startup of Memgraph, or at runtime.
+The `session_uuid` is the string value retrieved from setting the session trace on.
+
+The default value for the query log directory is empty so the user needs to set it either during startup or at runtime.
+
+If the `--query-log-directory` flag is empty, the user will receive the following error message:
```

Comment:
Should we add this flag to the `pages/database-management/configuration.mdx` file as well?

Reply:
I pushed a commit which does this